tests/drive_failure/drive_failure_test.go

```diff
@@ -0,0 +1,102 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/portworx/torpedo/drivers/node"
+	. "github.com/portworx/torpedo/tests"
+)
+
+func TestDriveFailure(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Torpedo: DriveFailure")
+}
+
+var _ = BeforeSuite(func() {
+	InitInstance()
+})
+
+var _ = Describe("Induce drive failure on of the nodes", func() {
+	testName := "drivefailure"
+	It("has to schedule apps and induce a drive failure on one of the nodes", func() {
+		var err error
+		contexts := ScheduleAndValidate(testName)
+
+		Step("get nodes for all apps in test and induce drive failure on one of the nodes", func() {
+			for _, ctx := range contexts {
+				var (
+					drives               []string
+					appNodes             []node.Node
+					nodeWithDrive        node.Node
+					driveToFail, driveID string
+				)
+
+				Step(fmt.Sprintf("get nodes where %s app is running", ctx.App.Key), func() {
+					appNodes, err = Inst().S.GetNodesForApp(ctx)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(appNodes).NotTo(BeEmpty())
+					nodeWithDrive = appNodes[0]
+				})
+
+				Step(fmt.Sprintf("get drive from node %v", nodeWithDrive), func() {
+					drives, err = Inst().V.GetStorageDevices(nodeWithDrive)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(drives).NotTo(BeEmpty())
+					driveToFail = drives[0]
```

Comment:
Why just the first drive and not all?

Reply:
Will add code to fail all the drives on one node.